modules/e2ee/Context.js

```diff
@@ -181,14 +194,20 @@ export class Context {
                 }
             }
 
-
             // If a signature is included, the S bit is set and a fixed number
             // of bytes (depending on the signature algorithm) is inserted between
             // CTR and the trailing byte.
-            const signatureLength = this._shouldSignFrame(encodedFrame) ? this._signatureOptions.byteLength : 0;
+            const signatureLength = this._shouldSignFrame(encodedFrame)
+                ? this._signatureOptions.byteLength + 1
+                    + ((this._frameSignatures.get(ssrc) || []).length * DIGEST_LENGTH[encodedFrame.type])
+                : 0;
+
             const frameTrailer = new Uint8Array(counterLength + signatureLength + 1);
 
-            frameTrailer.set(new Uint8Array(counter.buffer, counter.byteLength - counterLength));
+            frameTrailer.set(new Uint8Array(counter.buffer, counter.byteLength - counterLength),
+                frameTrailer.byteLength - (1
+                    + (signatureLength ? this._signatureOptions.byteLength : 0)
+                    + counterLength));
```

Comment:
the reason this is so complicated is that we do
data, signature, additional signatures, counter, signature, trailing byte
If we do this instead
data, signature, additional signatures, signature, counter, trailing byte
it gets easier